lib/sparse/src/index/inverted_index/inverted_index_ram.rs

```diff
@@ -68,16 +78,23 @@ impl InvertedIndexRam {
                     posting.upsert(posting_element);
                 }
                 None => {
-                    // resize postings vector
+                    // resize postings vector (fill gaps with empty posting lists)
                     self.postings.resize_with(dim_id + 1, PostingList::default);
                     // initialize new posting for dimension
                     self.postings[dim_id] = PostingList::new_one(id, weight);
                 }
             }
         }
+        // update indexed vector count by guessing whether we already have seen this point id
+        // given that there are no holes in the internal ids and that we are not deleting from the index
+        // we can just use the id as a proxy the count
+        if self.indexed_vector_count() < id as usize {
+            self.indexed_vector_count += 1;
```

Comment:
with `self.indexed_vector_count() < id` condition we cannot increment counter, we have to take `max`:
```
self.indexed_vector_count = max(self.indexed_vector_count, id);
```
Because counter will brake if you upsert points in non-sorted order